core/templates/pages/exploration-editor-page/services/exploration-rights.service.spec.ts

```diff
@@ -343,6 +320,22 @@ describe('Exploration rights service', () => {
     expect(ers.voiceArtistNames).toEqual([]);
   }));
 
+  it('should reject handler when saving a voice artist fails', fakeAsync(() => {
+    spyOn(
+      explorationRightsBackendApiService,
+      'assignVoiceArtistRoleAsyncPostData').and.returnValue(
+      Promise.reject());
+
+    ers.assignVoiceArtistRoleAsync('voiceArtist').then(
+      successHandler, failHandler);
+    tick();
+
+    expect(
+      explorationRightsBackendApiService
+        .assignVoiceArtistRoleAsyncPostData
+    ).toHaveBeenCalled();
```

Comment:
Please include the below test in the code

```
    expect(alertsService.addWarning).toHaveBeenCalledWith(
      '{{text}}'
    );
```

Reply:
@gp201 The problem I am facing here is since the warning message is raised from the backend and then  displayed on the frontend. While writing test for it, alertsService is not getting called, but the request is rejected. The same has been applied to [saveRoleChanges](https://github.com/oppia/oppia/pull/17809/files/73024a24b8a03033fd45151ac99b70292b2843cf#diff-7ff195e834d9cebdf69d4dd1b873942a6857329eb9ff459549e10348469e7849L108) in the same file.

If I am missing out on anything, It'll be great if you guide me in the right direction.